steward/src/allocation.rs

```diff
@@ -181,3 +205,44 @@ pub fn to_allocation(
             .collect(),
     }
 }
+
+pub async fn direct_rebalance(
+    cellar_address: H160,
+    tick_weight: Vec<TickWeight>,
+) -> Result<(), Error> {
+    let mut tick_info: Vec<CellarTickInfo> = Vec::new();
+    let config = APP.config();
+    let keystore = path::Path::new(&config.keys.keystore);
+    let keystore = FsKeyStore::create_or_open(keystore).expect("Could not open keystore");
+
+    let name = &config
+        .keys
+        .rebalancer_key
+        .parse()
+        .expect("Could not parse name");
+    let key = keystore.load(name).expect("Could not load key");
+
+    let key = key
+        .to_pem()
+        .parse::<k256::elliptic_curve::SecretKey<k256::Secp256k1>>()
+        .expect("Could not parse key");
+
+    let wallet: LocalWallet = Wallet::from(key);
+
+    let eth_host = config.ethereum.rpc.clone();
+    let client = Provider::<Http>::try_from(eth_host.clone()).unwrap();
+    let client = SignerMiddleware::new(client, wallet.clone());
+    let client = Arc::new(client);
+    let mut contract_state = UniswapV3CellarState::new(cellar_address, client);
+    for tick_weight in tick_weight {
+        if tick_weight.weight > 0 {
+            tick_info.push(from_tick_weight(tick_weight.clone()))
+        }
+    }
+    if std::env::var("CELLAR_DRY_RUN").expect("Expect CELLAR_DRY_RUN var") == "TRUE" {
+        Ok(())
+    } else {
+        tick_info.reverse();
```

Comment:
One of the first things that `contract_state.rebalance(tick_info)` does is call `reverse` on `tick_info`. We should remove this call or it will be reversed and then reversed again.

Reply:
done! https://github.com/PeggyJV/steward/pull/55/commits/20de02ea777ef1b44f1a7f98e709434beab179ff